.github/workflows/github-actions.yml

```diff
@@ -145,6 +145,34 @@ jobs:
       GIT_SECRET: ${{ secrets.GIT_SECRET }}
       PUBLISH: ${{ secrets.PUBLISH }}
     steps:
+      - name: Checkout Ambry
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+
+      - name: Cleanup Gradle Cache
+          # Remove some files from the Gradle cache which might have cached by earlier builds.
+        run: |
+          rm -f ~/.gradle/caches/modules-2/modules-2.lock
+          rm -f ~/.gradle/caches/modules-2/gc.properties
+
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
       - name: Publish To JFrog artifactory
         if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' && env.PUBLISH == 'true'}}
```

Comment:
hmm, i think i actually have to move this if block to apply to all of these

Reply:
GH actions supports if conditions at the job level, so using that: https://github.blog/changelog/2019-10-01-github-actions-new-workflow-syntax-features/

Reply:
Turns out that neither `env` or `secrets` is accessible from a job level conditional, so for now we have to duplicate the if statement on each step: https://github.community/t/conditional-job-using-environment-variable/141995